ASAManager/Helpers.cs

```diff
@@ -0,0 +1,237 @@
+                    null,
+                    token,
+                    null,
+                    null);
+                token = results.ContinuationToken;
+                foreach (var item in results.Results)
+                {
+                    if (item.Uri.AbsolutePath.Contains(fileName))
+                    {
+                        return true;
+                    }
+                }
+            } while (token != null);
+            
+            return false;
+        }
+
+        /**
+         * Get the blob storage connection string by querying
+         * environment variables.
+         */
+        public static string GetStorageConnectionString()
+        {
+            string blobStorageAccount = Environment.GetEnvironmentVariable(Constants.AZUREBLOB_ACCOUNT_ENV_VAR);
+            string blobStorageKey = Environment.GetEnvironmentVariable(Constants.AZUREBLOB_KEY_ENV_VAR);
+            string blobStorageEndpoint = Environment.GetEnvironmentVariable(Constants.AZUREBLOB_ENDPOINT_ENV_VAR);
+            return $"DefaultEndpointsProtocol=https;AccountName={blobStorageAccount};AccountKey={blobStorageKey};EndpointSuffix={blobStorageEndpoint}";
+        }
+
+        /**
+         * Returns true if the given blob has a file with the
+         * given filename between start and now, and that the
+         * file contains the given search string.
+         */
+        public static bool CheckIfBlobExistsAndContainsString(
+            DateTime start,
+            string blobStorageConnectionString,
+            string containerName,
+            string fileName,
+            string searchString)
+        {
+            return CheckIfBlobExistsAndIsValid(
+                start,
+                blobStorageConnectionString,
+                containerName,
+                fileName,
+                searchString,
+                true);
+        }
+
+        /**
+         * Returns true if the given blob has a file with the
+         * given filename between start and now, and that the
+         * file does not contain the given search string.
+         */
+        public static bool CheckIfBlobExistsAndDoesNotContainString(
+            DateTime start,
+            string blobStorageConnectionString,
+            string containerName,
+            string fileName,
+            string searchString)
+        {
+            return CheckIfBlobExistsAndIsValid(
+                start,
+                blobStorageConnectionString,
+                containerName,
+                fileName,
+                searchString,
+                false);
+        }
+
+        /**
+         * Send GET request using given httpClient against given requestAddress.
+         * Retry up to RETRY_COUNT times on non-OK status code, waiting
+         * RETRY_PAUSE_MSEC between requests. Return http response on success,
+         * null on failure.
+         */
+        public static IHttpResponse SendHttpRequestWithRetry(
```

Comment:
Wondering if it makes sense to move this to top-level Helpers so other projects can benefit.